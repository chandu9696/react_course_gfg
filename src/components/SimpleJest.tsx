import { useState } from "react"

export default function SimpleJest()
{
    const [name,Setname]=useState<string>('');
    return(
        <>
        <input type='text' onChange={(e)=>Setname(e.target.value)} data-testid='input'></input>
        <label data-testid='output' id='4'>{name}</label>

        </>
    )
}