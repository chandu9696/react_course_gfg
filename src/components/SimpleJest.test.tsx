import { fireEvent, render } from "@testing-library/react"
import SimpleJest from "./SimpleJest"

describe("SimpleJest",()=>{
    it('render without any crash',()=>{
        const {baseElement}=render(<SimpleJest/>)
        expect(baseElement).toBeInTheDocument()

    })
    it('render without any crash',()=>{
        const {getByTestId}=render(<SimpleJest/>)
        const inputElement=getByTestId("input")
        fireEvent.change(inputElement,{target:{value:'test'}})
        const outputElement=getByTestId("output")
        expect(outputElement.innerHTML).toBe("test")

        // expect(baseElement).toBeInTheDocument()

    })
    it('snapshot',()=>{
        const {baseElement}=render(<SimpleJest/>)
        expect(baseElement).toMatchSnapshot()

        // expect(baseElement).toBeInTheDocument()

    })
})