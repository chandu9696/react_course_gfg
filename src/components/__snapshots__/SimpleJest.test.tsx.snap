// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleJest snapshot 1`] = `
<body>
  <div>
    <input
      data-testid="input"
      type="text"
    />
    <label
      data-testid="output"
      id="4"
    />
  </div>
</body>
`;
